lib/Basic/LaneBasedExecutionQueue.cpp

```diff
@@ -241,7 +241,7 @@ class LaneBasedExecutionQueue : public ExecutionQueue {
   /// Returns the number of allowed foreground and background tasks.
   static auto estimateTaskLimits(unsigned numLanes) -> std::pair<unsigned, unsigned> {
     llbuild_rlim_t curOpenFileLimit = llbuild::basic::sys::getOpenFileLimit();
-    const unsigned reservedFileCount = (STDERR_FILENO+1) + 2 /* Database */
+    const unsigned reservedFileCount = (fileno(stderr) + 1) + 2 /* Database */
```

Comment:
1. This code estimates the number of file descriptors used up by the current process. The three standard file descriptors (stdin, stdout, stderr) utilize three file descriptors. I didn't want to just put 3 in there; wanted to hint at how the value `3` is computed by using the largest file descriptor and adding 1 (since STDIN_FILENO is 0).
2. We don't care about the value of the `stderr` file descriptor in this piece of resource estimation code. Even if it is 42 we still use a single file descriptor for each one of {stdin, stdout, stderr{.
3. https://codereview.qt-project.org/c/qt/qt/+/80277 (via @jakepetroules)

If the reason for this code not to compile on Windows is unavailability of `STDERR_FILENO` the best solution seems to just put `3` instead and comment appropriately.